File: app/src/main/java/com/apps/darkstorm/swrpg/StarWars/Stuff/Item.java
package com.apps.darkstorm.swrpg.StarWars.Stuff;

import java.util.ArrayList;

public class Item{
    //Version 1 0-2
    public String name;
    public String desc;
    public int count;
    public Object serialObject(){
        ArrayList<Object> tmp = new ArrayList<>();
        tmp.add(name);
        tmp.add(desc);
        tmp.add(count);
        return tmp.toArray();
    }
    public void loadFromObject(Object obj){
        Object[] tmp = (Object[])obj;
        switch (tmp.length){
            case 3:
                name = (String)tmp[0];
                desc = (String)tmp[1];
                count = (int)tmp[2];
        }
    }
    public boolean equals(Object obj){
        if (!(obj instanceof Item))
            return false;
        Item in = (Item)obj;
        return in.name.equals(name) && in.desc.equals(desc) && in.count == count;
    }
}

File: app/src/main/java/com/apps/darkstorm/swrpg/assistant/swElements/Item.kt
package com.apps.darkstorm.swrpg.assistant.swElements

import android.util.JsonReader
import android.util.JsonToken
import android.util.JsonWriter
import com.apps.darkstorm.swrpg.assistant.saveLoad.JsonSavable

data class Item(var name: String = "", var desc: String = "", var count: Int = 1, var encum: Int = 0): JsonSavable(){
    override fun save(jw: JsonWriter) {
        jw.beginObject()
        jw.name("name").value(name)
        jw.name("description").value(desc)
        jw.name("count").value(count)
        jw.name("encumbrance").value(encum)
        jw.endObject()
    }

    override fun load(jr: JsonReader) {
        jr.beginObject()
        while(jr.hasNext() && jr.peek() != JsonToken.END_OBJECT){
            if(jr.peek()!=JsonToken.NAME){
                jr.skipValue()
                continue
            }
            val jName = jr.nextName()
            when(jName){
                "name"-> name=jr.nextString()
                "description"-> desc = jr.nextString()
                "count"-> count = jr.nextInt()
                "encumbrance"-> encum = jr.nextInt()
            }
        }
        jr.endObject()
    }

    override fun clone() = copy()

}